widgets/src/components/common/WidgetBox.js

```diff
@@ -1,23 +1,66 @@
-import React from 'react';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
 import withStyles from '@material-ui/core/styles/withStyles';
 import classNames from 'classnames';
-import Paper from '@material-ui/core/Paper';
+import { Paper, Box, Typography, IconButton, Divider } from '@material-ui/core';
+import { ExpandMore as ExpandMoreIcon, ExpandLess as ExpandLessIcon } from '@material-ui/icons';
 
 const styles = {
   widgetBox: {
     background: '#FEFEFE',
-    padding: '20px 25px',
   },
 };
 
-const WidgetBox = ({ classes, passedClassName, children }) => (
-  <Paper variant="outlined" square className={classNames(classes.widgetBox, passedClassName)}>
-    {children}
-  </Paper>
-);
+const WidgetBox = ({
+  title,
+  actions,
+  collapsible,
+  hasDivider,
+  classes,
+  passedClassName,
+  children,
+}) => {
+  const [expanded, setExpanded] = useState(true);
+
+  const handleExpandClick = () => {
+    setExpanded(!expanded);
+  };
+
+  let renderedTitle = title;
+  if (typeof title === 'string') {
+    renderedTitle = <Typography variant="h5">{title}</Typography>;
+  }
+
+  return (
+    <Paper variant="outlined" square className={classNames(classes.widgetBox, passedClassName)}>
+      {title && (
+        <Box
+          display="flex"
+          justifyContent="space-between"
+          alignItems="center"
+          p={`${collapsible ? 8 : 20}px 25px`}
+        >
+          {renderedTitle}
```

Comment:
can't it simply be?
```<Typography variant="h5">{title}</Typography>;```

Reply:
Some widgets use a different Typography variant (e.g. h3, h2) and also some style their own title component. So that's why it has a condition `if (typeof title === 'string)' to render it with default "h5" variant only when the passed in title prop is a string.

Reply:
humm.. all WidgetBox title should have the same style in my opinion, but lez keep it until we get it defined by design 👍